src/core/Templater.ts

```diff
@@ -399,35 +399,53 @@ export class Templater {
-        // Not a problem with Calendar however since it creates the file with the existing content
-        await delay(300);
-
-        if (
-            file.stat.size == 0 &&
-            templater.plugin.settings.enable_folder_templates
-        ) {
-            const folder_template_match =
-                templater.get_new_file_template_for_folder(file.parent);
-            if (!folder_template_match) {
-                return;
-            }
+        await new Promise<void>((resolve) => {
+            let timeout: NodeJS.Timeout;
+            const onFileOpenRef = templater.app.workspace.on("file-open", async (openedFile: TAbstractFile) => {
+                if (openedFile !== file) {
+                    return;
+                }
+                clearTimeout(timeout);
+                templater.app.workspace.offref(onFileOpenRef);
+
+                // TODO: find a better way to do this
+                // Currently, I have to wait for the daily note plugin to add the file content before replacing
+                // Not a problem with Calendar however since it creates the file with the existing content
+                await delay(300);
+
+                if (
+                    file.stat.size == 0 &&
+                    templater.plugin.settings.enable_folder_templates
+                ) {
+                    const folder_template_match =
+                        templater.get_new_file_template_for_folder(file.parent);
+                    if (!folder_template_match) {
+                        resolve();
+                        return;
+                    }
 
-            const template_file: TFile = await errorWrapper(
-                async (): Promise<TFile> => {
-                    return resolve_tfile(templater.app, folder_template_match);
-                },
-                `Couldn't find template ${folder_template_match}`
-            );
-            // errorWrapper failed
-            if (template_file == null) {
-                return;
-            }
-            await templater.write_template_to_file(template_file, file);
-        } else {
-            await templater.overwrite_file_commands(file);
-        }
+                    const template_file: TFile = await errorWrapper(
+                        async (): Promise<TFile> => {
+                            return resolve_tfile(templater.app, folder_template_match);
+                        },
+                        `Couldn't find template ${folder_template_match}`
+                    );
+                    // errorWrapper failed
+                    if (template_file == null) {
+                        resolve();
+                        return;
+                    }
+                    await templater.write_template_to_file(template_file, file);
+                } else {
+                    await templater.overwrite_file_commands(file);
+                }
+                resolve();
+            });
+            timeout = setTimeout(() => {
```

Comment:
I wonder if we update the `delay` utility function to take an optional function as a second argument. then this could be `delay(300, timeOutFunc)`. Not a need to have, but might be a useful abstraction for other parts of the codebase... could come in a separate PR.